### PR TITLE
BUG/API: getitem behavior with list match ndarray/index/series

### LINKED ISSUE #33642: API: Series[index_with_no_matches] vs Series[list_with_no_matches]
We treat list indexers differently from array-like indexers:

```
ser = pd.Series(["A", "B"])
key = pd.Series(["C"])

>>> ser[key]
C    NaN
dtype: object

>>> ser[pd.Index(key)]
C    NaN
dtype: object

>>> ser[np.array(key)]
C    NaN
dtype: object

>>> ser[list(key)]
Traceback (most recent call last):
[...]
  File "/Users/bmendel/Desktop/pd/pandas/pandas/core/indexing.py", line 1312, in _validate_read_indexer
    raise KeyError(f"None of [{key}] are in the [{axis_name}]")
KeyError: "None of [Index(['C'], dtype='object')] are in the [index]"
```

Also inconsistent because `ser.loc[key]` raises for all 4 cases.

Is there a compelling reason for this?  I tried making all of these behave like the list case and only one test broke (that test being the example above).  The test was added in #5880.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #33642
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/indexing/test_getitem.py [hunk 1]
```diff
@@ -78,6 +78,18 @@ def test_getitem_median_slice_bug(self):
 
 
 class TestSeriesGetitemListLike:
+    @pytest.mark.parametrize("box", [list, np.array, pd.Index, pd.Series])
+    def test_getitem_no_matches(self, box):
+        # GH#33462 we expect the same behavior for list/ndarray/Index/Serioes
```
Comment (mroeschke) [inline pandas/tests/series/indexing/test_getitem.py:83]: Nit: `Serioes` -> Series

**Review by mroeschke — APPROVED** [review]
Looks like this closes https://github.com/pandas-dev/pandas/issues/33642 instead (the OP references the pull request for some reason)

**Comment by jbrockmendel** [comment]
updated, thanks

**Comment by jreback** [comment]
thanks

**Pull request merged**